cmd/bosun/web/roles.go

```diff
@@ -0,0 +1,89 @@
+package web
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/MiniProfiler/go/miniprofiler"
+	"github.com/captncraig/easyauth"
+)
+
+const (
+	canViewDash easyauth.Role = 1 << iota
+	canViewConfig
+	canPutData
+	canPerformActions
+	canRunTests
+	canSaveConfig
+	canViewAnnotations
+	canCreateAnnotations
+	canSilence
+	canManageTokens
+)
+
+const (
+	fullyOpen  easyauth.Role = 0
+	roleReader               = canViewDash | canViewConfig | canViewAnnotations
+	roleAdmin                = 0xFFFFFFFF
+)
+
+var roleDefs = &roleMetadata{
+	Permissions: []bitDesc{
+		{canViewDash, "View Dashboard", "Can view dashboard and alert state data, metrics, and graphs"},
+		{canViewConfig, "View Config", "Can view bosun configuration page"},
+		{canPutData, "Put Data", "Can put and index opentsdb data and metadata"},
+		{canPerformActions, "Actions", "Can acknowlege and close alerts"},
+		{canRunTests, "Run Tests", "Can execute expressions, graphs, and rule tests"},
+		{canSaveConfig, "Save Config", "Can alter and save bosun rule config"},
+		{canViewAnnotations, "View Annotations", "Can view annotations on graph page"},
+		{canCreateAnnotations, "Create Annotations", "Can add and manage annotations via api"},
+		{canSilence, "Silence", "Can add and manage silences"},
+		{canManageTokens, "Manage Tokens", "Can manage authorization tokens"},
+	},
+	Roles: []bitDesc{
+		{roleReader, "Reader", "Read access to dashboard"},
+		{roleAdmin, "Admin", "Full access to all functionality"},
+	},
+}
+
+type bitDesc struct {
+	Bits easyauth.Role
+	Name string
+	Desc string
+}
+type roleMetadata struct {
+	Permissions []bitDesc
+	Roles       []bitDesc
+}
+
+func parseRole(s string) (easyauth.Role, error) {
```

Comment:
This is just for converting roles in the config file to our numeric values. Should never be used at runtime. Maybe this should be somewhere in the config package or something?